### DISCUSSION TITLE
Possibility to upload OpenAPI definiton from URL

### DISCUSSION THREAD

**Discussion by eshepelyuk** [Apicurio/apicurio-registry#1494] [Ideas]
Hello, 
As far as I could see - currently it's only possible to upload OpenAPI definition via file.
It could be great addition to be able to provide URL, so registry could download definition on it's own.

**Comment by EricWittmann** [comment]
I think this is a reasonable thing to do.  It shouldn't be particularly difficult to implement.  The risk of course is that the feature might not work at all in some deployment configurations (where the server cannot access external URLs).  But I think that's just a documentation issue really.

@Apicurio/developers Thoughts?

**Reply by jsenko** [reply]
I agree. The only issue I can think of is a security one, e.g. if the URL is malicious and points to an extremely large file that could cause memory depletion. Unlikely, but the download has to be done carefully.

**Reply by carlesarnal** [reply]
+1. I agree with Jakub, the only issue that I see right now is security.

**Reply by eshepelyuk** [reply]
Well, I could propose following workaround to reduce risks

1. Registry issues `HTTP HEAD` request to the target URL.
1. According to HTTP specification, target server must return `Content-Length` header.
1. Registry analyzes value of `Content-Length` to check if file with declared size could be processed securely.
1. Finally, registry issues `HTTP GET` to the target URL and fetches only amount of bytes specified by `HTTP HEAD` from step 1.

WDYT ?

PS: the proposal implies that target URLs **must** support `HTTP HEAD` method.

**Comment by jsenko** [comment]
I've created an issue for this https://github.com/Apicurio/apicurio-registry/issues/1499
Thanks @eshepelyuk
